### PR TITLE
Parsing html2

### PR DESCRIPTION
**Pull request by jtydhr88**

### CODE REVIEW

File: tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java [hunk 1]
```diff
@@ -58,4 +61,10 @@ public UpgradePlan newUpgradePlan(
 
 	public void startUpgradePlan(UpgradePlan upgradePlan);
 
+	public void updateUpgradePlanOutlines(List<String> outlines);
+
+	public final List<String> defaultUpgradePlanOutlines = Arrays.asList(
+		"liferay-code-upgrade-plan," +
+			"http://localhost:8080/web/guest/docs/7-2/reference#upgrading-code-to-liferay-portal-ce-72");
```
Comment (jtydhr88) [inline tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java]: we have to set an internal url until we get a public office link. And I add the preference page to allow to set custom outline link.

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
hey @jtydhr88  I'd like to redesign how this works as follows:

1. UpgradePlanner should not know where all the upgrade plans are, it should just be able to load any upgrade plan that is given.  So the "newUpgradePlan" should take a URL as the final parameter that will be the URL of the upgrade plan content
2. we don't need `loadAllUpgradePlanOutlines` and `updateUpgradePlans` those should be stored in another location, the same place were the preference page changes are stored, likely in the UpgradePlanUI Plugin class.

File: tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java [hunk 2]
```diff
@@ -37,6 +38,8 @@
 
 	public UpgradePlan getCurrentUpgradePlan();
 
+	public List<String> loadAllUpgradePlanOutlines();
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java]: Lets not add this method, see previous comment.

File: tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java [hunk 3]
```diff
@@ -58,4 +61,10 @@ public UpgradePlan newUpgradePlan(
 
 	public void startUpgradePlan(UpgradePlan upgradePlan);
 
+	public void updateUpgradePlanOutlines(List<String> outlines);
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java]: Shouldn't be needed.

File: tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java [hunk 4]
```diff
@@ -58,4 +61,10 @@ public UpgradePlan newUpgradePlan(
...
+
+	public final List<String> defaultUpgradePlanOutlines = Arrays.asList(
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.upgrade.plan.core/src/com/liferay/ide/upgrade/plan/core/UpgradePlanner.java]: Lest store this in UpgradePlannUIPlugin

**Pull request closed without merging**